packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -326,16 +325,26 @@ abstract class Layer extends AbstractNode with DiagnosticableTreeMixin {
     AnnotationResult<S> result,
     Offset localPosition, {
     @required bool onlyFirst,
-  });
+  }) {
+    debugPrint(
+      'Layer $runtimeType has not implemented Layer.findAnnotations. '
+      'If the layer class does not care about annotations, override '
+      'findAnnotations with a simple "return false". '
+      'If it overrides find and findAll to add custom behaviors, '
+      'please change to overriding Layer.findAnnotations instead, '
+      'otherwise these behaviors will no longer take effect. '
+      'See https://github.com/flutter/flutter/pull/37896 for details.');
+    return false;
```

Comment:
we should never print warnings. either throw, or don't print.

Reply:
Removed